core/src/main/java/io/questdb/cairo/SymbolMapWriter.java

```diff
@@ -155,6 +158,7 @@ public int getSymbolCount() {
     public int put(CharSequence symbol) {
 
         if (symbol == null) {
+            offsetMem.putBool(HEADER_NULL_FLAG, true);
```

Comment:
this write should be cached i feel like, e.g. read of class member is faster than write to memory mapped file

Reply:
yes, working on that, just wanted to push the latest changes